test/responses.js

```diff
@@ -662,29 +670,131 @@ suite('Responses', function () {
       });
     });
 
-    test('Deleting a response', function (done) {
+    // Per-test setup.
+    setup(function () {
+      // Create 3 entries, two of which will be part of the same Response doc.
+      var parcelBase = Math.floor(10000 * Math.random() + 5000);
+      var data = fixtures.makeResponses(2, {
+        parcelBase: parcelBase
+      });
+      this.rawResponses = data.responses.concat(fixtures.makeResponses(1, {
+        parcelBase: parcelBase
+      }).responses);
+
+      return Promise.bind(this)
+      .then(function () {
+        return this.rawResponses;
+      }).map(function (response) {
+        return request.postAsync({
+          url: BASEURL + '/surveys/' + surveyId + '/responses',
+          json: { responses: [response] },
+          jar: ownerJar
+        }).spread(function (response, body) {
+          return body.responses[0];
+        });
+      }, {
+        concurrency: 1
+      }).then(function (responses) {
+        this.responses = responses;
+      });
+    });
 
-      // Delete the response.
-      request.del({
+    test('Deleting one entry for a base object that has two entries', function () {
+      var id = this.responses[0].id;
+      var count;
+      return request.getAsync({
+        url: BASEURL + '/surveys/' + surveyId + '/responses?startIndex=0&count=10000',
+        jar: ownerJar
+      }).spread(function (response, body) {
+        var data = JSON.parse(body);
+        count = data.responses.length;
+
+        // Delete the response.
+        return request.delAsync({
           url: BASEURL + '/surveys/' + surveyId + '/responses/' + id,
           jar: ownerJar
-        },
-        function(error, response) {
-          should.not.exist(error);
-          should.exist(response);
-          response.statusCode.should.equal(204);
+        });
+      }).spread(function(response) {
+        should.exist(response);
+        response.statusCode.should.equal(204);
 
-          // Try to get the response
-          request.get({
-            url: BASEURL + '/surveys/' + surveyId + '/responses/' + id,
-            jar: ownerJar
-          }, function (error, response) {
-            should.not.exist(error);
-            response.statusCode.should.equal(404);
-            done();
-          });
-        }
-      );
+        // Try to get the response
+        return request.getAsync({
+          url: BASEURL + '/surveys/' + surveyId + '/responses/' + id,
+          jar: ownerJar
+        });
+      }).spread(function (response) {
+        response.statusCode.should.equal(404);
+
+        // Confirm that we have the correct number of responses after a
```

Comment:
In the spirit of testing one thing at a time, I think this should be split out into a separate test. I also see how it fits directly into the setup and intent of this test. What do you think?